build_scripts/windows/scripts/build.cmd

```diff
@@ -137,18 +144,25 @@ if %errorlevel% neq 0 goto ERROR
 
 pushd %BUILDING_DIR%
 %BUILDING_DIR%\python.exe %REPO_ROOT%\scripts\compact_aaz.py
+if %errorlevel% neq 0 goto ERROR
 %BUILDING_DIR%\python.exe %~dp0\patch_models_v2.py
+if %errorlevel% neq 0 goto ERROR
 %BUILDING_DIR%\python.exe %REPO_ROOT%\scripts\trim_sdk.py
+if %errorlevel% neq 0 goto ERROR
 popd
 
 REM Remove pywin32 help file to reduce size.
 del %BUILDING_DIR%\Lib\site-packages\PyWin32.chm
 
 echo Creating the wbin (Windows binaries) folder that will be added to the path...
 mkdir %BUILDING_DIR%\wbin
-copy %REPO_ROOT%\build_scripts\windows\scripts\az.cmd %BUILDING_DIR%\wbin\
-copy %REPO_ROOT%\build_scripts\windows\scripts\azps.ps1 %BUILDING_DIR%\wbin\
-copy %REPO_ROOT%\build_scripts\windows\scripts\az %BUILDING_DIR%\wbin\
+if "%TARGET%"=="msi" (
+    copy %REPO_ROOT%\build_scripts\windows\scripts\az_msi.cmd %BUILDING_DIR%\wbin\az.cmd
+    copy %REPO_ROOT%\build_scripts\windows\scripts\azps.ps1 %BUILDING_DIR%\wbin\
+    copy %REPO_ROOT%\build_scripts\windows\scripts\az %BUILDING_DIR%\wbin\
+) else (
+    copy %REPO_ROOT%\build_scripts\windows\scripts\az_zip.cmd %BUILDING_DIR%\wbin\az.cmd
```

Comment:
Other Microsoft-owned apps mentioned in https://github.com/Azure/azure-cli/pull/27911#discussion_r1454996530 put their executables such as `Code.exe`, `pwsh.exe` under the root folder. Perhaps we can do the same?

But this approach has its own problem: adding Azure CLI's directory to `PATH` will also put `python.exe` under `PATH`. If a user accidentally executes `python.exe`, that may corrupt Azure CLI's virtual environment.

Reply:
Why do we use `wbin`, instead of `bin`?

Reply:
Oh. The explanation is right in the comment: 

https://github.com/Azure/azure-cli/blob/04d82da1bf0a319ed38a9eabe7716119f50e402d/build_scripts/windows/scripts/build.cmd#L147-L148

Reply:
Hmm, this package is for Windows, `w` seems redundant.

Reply:
> Other Microsoft-owned apps mentioned in [#27911 (comment)](https://github.com/Azure/azure-cli/pull/27911#discussion_r1454996530) put their executables such as `Code.exe`, `pwsh.exe` under the root folder. Perhaps we can do the same and put `az.cmd` under the root folder?
> 
> But this approach has its own problem: adding Azure CLI's folder to `PATH` will also put `python.exe` under `PATH`. If a user accidentally executes `python.exe`, that may corrupt Azure CLI's virtual environment.

Adding python in the path could also create instability if the platform (for example a custom build agent) is using python for other purposes.

I think that `wbin` will be confusing and the information it adds is not necessary.

Reply:
@dcaro I can change the folder to `bin` in zip package.
Should we also do this in MSI? We never mention `wbin` in doc, and it's automatically added into PATH during MSI installation. This change should not affect users unless they use full path to run `az`.

Reply:
let's leave the folder name in the msi as is for now, as you mention, it is automatically added to path and transparent to users.